src/Engine/Component/Script.hpp

```diff
@@ -32,18 +33,77 @@ namespace Component {
             
             /// The instance of the script class.
             asIScriptObject* instance = nullptr;
-      
-            /// Entities referenced by this script.
-            std::vector<Entity*> refList;
 
-            ///Map containing the properties, maps a pair of a value and it's type to a name (map<nameOfProperty, pair<typeOfProperty, valueOfProperty>>)
-            std::map<std::string, std::pair<int, void*>> propertyMap;
+            /// Add a property to the propertyMap.
+            /**
+             * @param name The name of the property.
+             * @param type The asTypeID of the property.
+             * @param size The size in number of bytes of the property.
+             * @param data A pointer to the data to store.
+             */
+            ENGINE_API void AddToPropertyMap(std::string name, int type, int size, void* data);
 
-            /// Fills the property map.
-            void FillPropertyMap();
+            /// Add a property to the propertyMap.
+            /**
+             * @param name The name of the property.
+             * @param type The asTypeID of the property.
+             * @param target A pointer that points where to copy the data.
+             */
+            ENGINE_API void CopyDataFromPropertyMap(std::string name, void* target);
+
+            /// Add a property to the propertyMap.
+            /**
+             * @param name The name of the property.
+             * @return A pointer to the data of the property.
+             */
+            ENGINE_API void* GetDataFromPropertyMap(std::string name);
+
+            /// Save the component.
+            /**
+             * @param name The name of the property.
+             * @param type The asTypeID of the property.
+             * @return Is there a property with the provided name and type in propertyMap.
+             */  
+            ENGINE_API bool isInPropertyMap(std::string name, int type);
 
             /// Clears the property map.
             ENGINE_API void ClearPropertyMap();
 
+        private:
+
+            class Property {
+                
+            public:
+
+                Property() {
+
+                    typeID = -1;
+                    size = -1;
+                    data = nullptr;
+
+                }
+
+                Property(int _typeID, int _size, void* _data) {
+
+                    typeID = _typeID;
+                    size = _size;
+                    data = malloc(size);
```

Comment:
While this works, the C++ way would be to use new/delete[], but then we need to cast the pointer to char and back so I'm split as to what to do here. Unless anyone else has anything to say I'm fine with malloc.